### ISSUE TITLE
Cannot filter (search) in Lookup and DataGrid columns on Android mobile/tablet

### ISSUE THREAD

**Issue by edgarinfor** [infor-design/enterprise-ng#1489]
**Describe the bug**
There is an existing browser inconsistency that occurs when the Filter (search) on the DataGrid columns is used on Android devices.

The device's on-screen keyboard displays different "action" key, either as "Next" or "Return", depending on the different browsers, which affects the filtering behavior preventing it to work in some cases.

- In case of Chrome, we are getting the "Next" button.
- In case of Firefox we are getting the "Return" button.

The different buttons, don't trigger the same component event which seems to cause the issue.

**To Reproduce**
The issue occurs on different IDS demo pages:
https://main-enterprise.demo.design.infor.com/components/lookup/example-clientside-paging-with-filtering.html
https://main-enterprise.demo.design.infor.com/components/lookup/example-filter-lookup.html

1. Open the demo pages on Android using different browsers (Chrome, Firefox)
2. Try to Filter (Search) by Column
3. See the inconsistency occur in the on-screen keyboard and behavior

**Expected behavior**
Although browsers inconsistencies are unavoidable, the column filter and search functionality should work as expected in all supported browsers.

We wonder if any new HTML attributes could make any difference, like this one:
https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint

**Version**
- ids-enterprise-ng: Current

**Screenshots**
![Chrome_Datagrid](https://github.com/infor-design/enterprise-ng/assets/39097500/722f2472-5b3f-43d8-9d24-c0116e948904)
![Chrome_Datagrid_Modal](https://github.com/infor-design/enterprise-ng/assets/39097500/47a4f7d6-ad82-4e20-8020-6b871bfef2eb)
![Chrome_Lookup](https://github.com/infor-design/enterprise-ng/assets/39097500/09251ddb-778a-4679-bdf8-1e174370f50a)
![Firefox_Lookup1](https://github.com/infor-design/enterprise-ng/assets/39097500/301b5746-324d-4789-a218-d3cc7bb37810)
![Firefox_Lookup2](https://github.com/infor-design/enterprise-ng/assets/39097500/b9597fc8-c21d-484c-a211-ab5c2500aa44)

**Platform**
 - Device: Android
 - Browser Name: Chrome, Safari

**Additional context**
We have advised customers to use an alternative browsers but, this is now an escalated issue.

**Comment by tmcconechy** [comment]
Yeah sounds like we could try adding https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint first.

Good Bug.

**Comment by tmcconechy** [comment]
@edgarinfor would you be able to test this out and see if https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint works by modifying the dom in the inspector.

Im also unclear what the value should be? "go" "enter"?

If you can put the test in place and tell me what works i can put the fix in?

**Comment by tmcconechy** [comment]
@edgarinfor actually can you clarify?

- lookup you want it to open the lookup on keyboard enter?
- datagrid should obviously filter on enter -> looks like it goes to the next field

I think we would have to make enter work on lookup then test that the keyboard can pass enter with 
they know they can click the button on the lookup right?

**Comment by edgarinfor** [comment]
**Tim**, for the lookup, I think that it should behave as the other fields.

That is, if you enter a value in the lookup filter then press enter, it should try to filter the datagrid by that column filter value.

It could be hard to know what the user intentions are with a lookup, so to decide whether to open the lookup or search instead it's really tricky.

Since there's an explicit way to open the dialog (e.g. clicking the icon), I think filtering on "enter" works well.

Does that makes sense?

The only case where it would make clear sense to open the dialog on "enter" is if the lookup can "auto resolve", meaning that if the value entered doesn't exist in the results (invalid) then, it will open the dialog so you actually have to look for a valid one.

That's basically how we implement lookups when used as regular fields (outside of the datagrid filters) but, that may require a little bit more work.

**Comment by tmcconechy** [comment]
@edgarinfor no that doesnt 100% clarify it for the lookup

- for filter row i fixed it it will now filter (enterkeyhint="enter")
- for lookup i could make it a setting (it is already with `attributes` as mentioned)
- but should we make the enterkey do the dialog opening? Or maybe just set it to "done" so it closes then they click the search icon?


So 
a) on lookup should it open the dialog on enter or
b) just close the virtual keyboard

**Comment by edgarinfor** [comment]
As discussed, we can leave the behavior of the standalone lookup fields as it is and, if they need to be changed, we could use the "attributes" settings on a case by case basis.
